src/ansible_builder/main.py

```diff
@@ -154,7 +154,9 @@ def build_command(self):
             command.extend(["-t", tag])
 
         for key, value in self.build_args.items():
-            if value:
+            if value == '':
```

Comment:
This logic should be wrapped up inside the `BuildArgAction` below, and probably just use `partition` to split the string- if the last element has no value but the second element is `=`, you know you need to tack an `=` onto the end of the key.

Also, as a general rule, embedding empty quotes in something that's going to get round-tripped to another CLI is probably going to cause a lot more problems than it solves; just having a trailing `=` on the arg should be sufficient in this case.

Reply:
On further reflection, the placement here is fine, but we definitely don't want the `''` in the final command for that case (just `{key}=`). The logic should probably be more explicit though, as the YAML layer *could* technically provide an integer `0` value (the JSON schema might currently prevent it, but I don't know for sure), which would erroneously trigger the `else`. 

So probably just to be super safe have `BuildArgAction` provide a value of `None` for eg `--build-arg FOO`, `''` for `--build-arg FOO=` and of course the real value for anything else... Then the logic here should explicitly check for value being `None` and `''`, then the unqualified `else` case is the `{key}={value}` one.